shremote.py

```
@@ -524,60 +547,44 @@ def __init__(self, program_name, cmd_cfg, index = None):
-
-
-        if 'enforce_duration' in self.program.cfg:
-            enforce_duration = self.program.cfg.formatted('enforce_duration', **self.dict())
-            if enforce_duration == True:
-                if self.duration is not None:
-                    self.enforced_duration = self.duration - 1
-                else:
-                    self.enforced_duration = enforce_duration
-            else:
-                self.enforced_duration = self.program.cfg.enforce_duration
-        else:
-            if self.duration is not None:
-                self.enforced_duration = self.duration - 1
-            else:
-                self.enforced_duration = None
+            self.duration = int(try_eval(duration_raw, 'duration'))
 
+        #Enforce duration can be specified in the program or in the command
+        self.enforced_duration = None
+        enforce_duration_loc = None
         if 'enforce_duration' in cmd_cfg:
-            enforce_duration = cmd_cfg.formatted('enforce_duration', **self.dict())
+            enforce_duration_loc = cmd_cfg
+        elif 'enforce_duration' in self.program.cfg:
+            enforce_duration_loc = self.program.cfg
+
+        #If specificed, default to self.duration or can take a value
+        if enforce_duration_loc is not None:
+            enforce_duration = enforce_duration_loc.formatted('enforce_duration', **self.dict())
             if enforce_duration == True:
                 if self.duration is None:
-                    log_fatal("Specified enforce duration = True with no duration specified")
+                    log_fatal(
+                        "{}: Specified enforce_duration=True with no duration specified".format(program_name)
+                    )
                 self.enforced_duration = self.duration - 1
-            else:
+            elif isinstance(enforce_duration, int) and enforce_duration > 0:
```

It seems that this might cause a situation where self.enforced_duration is not set at all. Like if enforce_duration is a float? Will that now break? Checking now

---

Do we want to super floats?

---

Yes? Why not? It does now, and this removes that support.
More so, I am afraid that this does not set enforced_duration at all. Rather than complaining loudly at this time, it would cause weird behavior later when self.enforced_duration does not exist

---

We can have
`elif (isinstance(enforce_duration, int) or isinstance(enforce_duration, float)) and enforce_duration > 0:`

To support both.

The point is to allow enforce_duration to be both a boolean and a number.

---

Sure, that's fine.
Use `isinstance(enforce_duration, (int, float))`

---

Also, a final `else` condition to make sure it didn't evaluate to another type, and should fail loudly if it did with a reasonable error.